File: Python/algorithmic_toolbox/17_majority_element.py
import sys
from collections import Counter


def convert_to_intlist(line):
    return [int(elem) for elem in line.split()]


def has_majority_elem(lst, length):
    counter = Counter(lst)
    _, count = counter.most_common(n=1)[0]
    if count > length // 2:
        return 1
    return 0


def main():
    data = sys.stdin.read().splitlines()
    n = int(data[0])
    lst = convert_to_intlist(data[1])
    result = has_majority_elem(lst, n)
    print(result)


if __name__ == "__main__":
    main()

File: Julia/algorithmic_toolbox/17-majority-element.jl
converttointlist(line) = map(x -> parse(Int, x), split(line))


function hasmajorityelem(list, n)
    counter = begin
        dict = Dict{Int, Int}()
        for x in list
            if haskey(dict, x)
                dict[x] += 1
            else
                dict[x] = 1
            end
        end
        dict
    end

    half = n ÷ 2
    for count in values(counter)
        if count > half
            return 1
        end
    end
    0
end


function main()
    n = parse(Int, readline())
    list = readline() |> converttointlist
    result = hasmajorityelem(list, n)
    println(result)
end


main()
